src/plain.jl

```diff
@@ -606,6 +606,27 @@ function h5open(f::Function, args...)
     end
 end
 
+if VERSION >= v"0.4"
+    function h5rewrite(f::Function, filename::AbstractString, args...)
+        tmppath,tmpio = mktemp(dirname(filename))
+        close(tmpio)
+
+        try
+            fid = h5open(tmppath, "w", args...)
+            val = try
+                ret = f(fid)
+            finally
+                close(fid)
```

Comment:
That wasn't the intention.  In case of exception, it should close the file, and the exception should propagate to the outer catch block which will delete it.